### ISSUE TITLE
5/6. The leaderboard/refer-a-mate should be administer-able.

### ISSUE THREAD

**Issue by christoshrousis** [decryption256/sizzle-refer-a-mate#5]
- [ ] Store a flag in the database that sets the state of the leaderboard. "frozen"
- [ ] When storing referral IDs and counts, we will need to ensure that referral counts are reset.

Needs a bit more clarity on what the definition of a competition is, and whether it's just a "turn it on, turn it off" thing.

My suggestion is storing a node the looks something like this:
```
results
|-- 2019-01-12 (datetime?? or maybe a competition id?)
     |-- referralid
          |-- List of emails     
```

**Comment by christoshrousis** [comment]
@decryption256 When you have a chance, can you give me a bit more detail on how you expect competitions to run?

Are you comfortable with the idea of logging into a firebase database and flipping a variable or two, or do you need me to build you an admin interface?

**Comment by christoshrousis** [comment]
Is there a reward to the top person, or is it just temporary fame?

**Comment by decryption256** [comment]
Yep, there's a reward for the top referrer that'll change each month as an incentive (e.g: Sonos One speaker, Apple Airpods, mechanical keyboard, Ring video doorbell, etc.).

No issues with me getting in to a DB and changing true to false or something like that, to start and stop the comp.

I'm planning for the competition to run for a full month - e.g: launch on the 1st of March (I'd email the list to announce), end on the 31st of March. Then on the 1st of April, I'd announce the winner of March and launch new competition for April that goes until the end of April, then announce the winner of that on the 1st of May and so on.
